plugins/inputs/burrow/burrow.go

```diff
@@ -0,0 +1,349 @@
+package burrow
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/errchan"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type ConsumerOffset struct {
+	Offset     int64 `json:"offset"`
+	Timestamp  int64 `json:"timestamp"`
+	Lag        int64 `json:"lag"`
+	Artificial bool  `json:"-"`
+}
+
+type PartitionStatus struct {
+	Topic     string         `json:"topic"`
+	Partition int32          `json:"partition"`
+	Status    string         `json:"status"`
+	Start     ConsumerOffset `json:"start"`
+	End       ConsumerOffset `json:"end"`
+}
+
+type ConsumerGroupStatus struct {
+	Cluster         string             `json:"cluster"`
+	Group           string             `json:"group"`
+	Status          string             `json:"status"`
+	Complete        bool               `json:"complete"`
+	Partitions      []*PartitionStatus `json:"partitions"`
+	TotalPartitions int                `json:"partition_count"`
+	Maxlag          *PartitionStatus   `json:"maxlag"`
+	TotalLag        uint64             `json:"totallag"`
+}
+
+type BurrowResponseRequestInfo struct {
+	URI     string `json:"url"`
+	Host    string `json:"host"`
+	Cluster string `json:"cluster"`
+	Group   string `json:"group"`
+	Topic   string `json:"topic"`
+}
+type HTTPResponseError struct {
+	Error   bool                      `json:"error"`
+	Message string                    `json:"message"`
+	Result  map[string]string         `json:"result"`
+	Request BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseClusterList struct {
+	Error    bool                      `json:"error"`
+	Message  string                    `json:"message"`
+	Clusters []string                  `json:"clusters"`
+	Request  BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseTopicList struct {
+	Error   bool                      `json:"error"`
+	Message string                    `json:"message"`
+	Topics  []string                  `json:"topics"`
+	Request BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseTopicDetail struct {
+	Error   bool                      `json:"error"`
+	Message string                    `json:"message"`
+	Offsets []int64                   `json:"offsets"`
+	Request BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseConsumerList struct {
+	Error     bool                      `json:"error"`
+	Message   string                    `json:"message"`
+	Consumers []string                  `json:"consumers"`
+	Request   BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseConsumerStatus struct {
+	Error   bool                      `json:"error"`
+	Message string                    `json:"message"`
+	Status  ConsumerGroupStatus       `json:"status"`
+	Request BurrowResponseRequestInfo `json:"request"`
```

Comment:
Probably would make sense to have a type:
```
type BurrowResponse struct {
	Error     bool                      `json:"error"`
	Message   string                    `json:"message"`
	Request   BurrowResponseRequestInfo `json:"request"`
}
```

Then you should be able to embed it like:
```
type BurrowResponseConsumerStatus struct {
	Status  ConsumerGroupStatus       `json:"status"`
	BurrowResponse
}

```